Pay/PaySession.swift

```diff
@@ -27,33 +27,36 @@
 import Foundation
 import PassKit
 
-public protocol PayControllerDelegate: class {
-    func payController(_ payController: PayController, didRequestShippingRatesFor address: PayPostalAddress, provide: @escaping (PayCheckout, [PayShippingRate]) -> Void)
-    func payController(_ payController: PayController, didSelectShippingRate shippingRate: PayShippingRate, provide: @escaping  (PayCheckout) -> Void)
-    func payController(_ payController: PayController, didCompletePayment token: Data, checkout: PayCheckout, completeTransaction: @escaping (PayController.TransactionStatus) -> Void)
-    func payControllerDidFinish(_ payController: PayController)
+public protocol PaySessionDelegate: class {
+    func paySession(_ paySession: PaySession, didRequestShippingRatesFor address: PayPostalAddress, checkout: PayCheckout, provide: @escaping (PayCheckout?, [PayShippingRate]) -> Void)
+    func paySession(_ paySession: PaySession, didSelectShippingRate shippingRate: PayShippingRate, checkout: PayCheckout, provide: @escaping (PayCheckout?) -> Void)
+    func paySession(_ paySession: PaySession, didAuthorizePayment authorization: PayAuthorization, checkout: PayCheckout, completeTransaction: @escaping (PaySession.TransactionStatus) -> Void)
+    
+    func paySessionDidFinish(_ paySession: PaySession)
 }
 
-public class PayController: NSObject {
+public class PaySession: NSObject {
     
     public enum TransactionStatus {
         case success
         case failure
     }
     
-    public weak var delegate: PayControllerDelegate?
+    public weak var delegate: PaySessionDelegate?
     
     public let merchantID: String
+    public let identifier: String
     
-    public fileprivate(set) var currency:      PayCurrency?
-    public fileprivate(set) var checkout:      PayCheckout?
-    public fileprivate(set) var shippingRates: [PayShippingRate]?
+    internal fileprivate(set) var currency:      PayCurrency?
+    internal fileprivate(set) var checkout:      PayCheckout?
```

Comment:
I don't think we should be afraid of force unwrapping. It's not inherintly "bad". In this case, we can be certain that any attempt to access `checkout` __after__ calling authorize payment method will always have a valid checkout. If doing so ever returns `nil` we most definitely want to crash because something has seriously gone wrong. An implicitly-unwrapped optional, as @davidmuzi suggested can be a more elegant solution.